core/src/mast/mod.rs

```diff
@@ -132,9 +135,141 @@ impl MastForest {
             self.roots.push(new_root_id);
         }
     }
+
+    /// Removes all nodes in the provided set from the MAST forest.
+    ///
+    /// It also returns the map from old node IDs to new node IDs; or `None` if the set of nodes to
+    /// remove was empty. Any [`MastNodeId`] used in reference to the old [`MastForest`] should be
+    /// remapped using this map.
+    pub fn remove_nodes(
```

Comment:
What would happen here if we try to remove nodes which are not "orphaned" (i.e., have a parent). Would we panic? Or would it silently result in an invalid `MastForest`?

In either case, we should describe in the comments what happens (e.g., by adding `# Panics` section).

Reply:
we would silently return an invalid `MastForest`; added that to docstring